### PR TITLE
Add hyperparameter tuning

### PR DESCRIPTION
**Pull request by aarushi-mahajan**
Added hyperparameter tuning script for the Lasso model (sklearn LogisticRegression L1/saga).

- Grid search over C values using 5-fold stratified CV
- PR-AUC as primary metric due to ~9:1 class imbalance
- Balanced sample weights applied per fold
- Threshold tuning done post-CV via precision-recall curve to maximise F1
- Processed matrices cached to disk to avoid redundant feature building on reruns

### CODE REVIEW

**Comment by alexisgrangier** [comment]
To Drew's intention: this branch is being worked on by several people over a long time. Code review happens 'live' as we collaboratively develop the code in this branch. We will be merging this branch with main at a later stage.
